### ISSUE TITLE
Breaking change in generated derive implementation not detected

### ISSUE THREAD

**Issue by weiznich** [obi1kenobi/cargo-semver-checks#533]
### Steps to reproduce the bug with the above code

git clone https://github.com/diesel-rs/diesel
cd diesel/diesel
cargo semver-checks --baseline-rev v2.1.0 --only-explicit-features --features postgres --features sqlite --features mysql --features extras --features with-deprecated --manifest-path .

### Actual Behaviour

```
     Cloning v2.1.0
     Parsing diesel v2.1.1 (current)
     Parsing diesel v2.1.0 (baseline)
    Checking diesel v2.1.0 -> v2.1.1 (patch change)
   Completed [   0.187s] 45 checks; 45 passed, 0 unnecessary
```

### Expected Behaviour

`cargo semver-check` does not report a breaking change for the given release, although it contains technically breaking changes. On of the derive macros was changed in such a way that it now generate additional trait implementations. Users could have written these implementations manually before updating to the new patch release. Such an existing implementation would result in a conflicting implementation error as the same implementation then would have existed twice. See https://github.com/diesel-rs/diesel/pull/3747 for reference. 

(I'm well aware that this is essentially impossible to detect with only the source code of the relevant crate, as it requires evaluating the derive macro for a specific input)

#### Software version

cargo-semver-checks 0.22.1

#### Operating system

Linux 6.4.7-200.fc38.x86_64

#### Command-line

```bash
/home/weiznich/.cargo/bin/cargo-semver-checks semver-checks --bugreport 
```

#### cargo version

```
> cargo -V 
cargo 1.68.0 (115f34552 2023-02-26)
```

#### Compile time information

- Profile: release
- Target triple: x86_64-unknown-linux-gnu
- Family: unix
- OS: linux
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-unknown-linux-gnu

**Comment by obi1kenobi** [comment]
Yeah, this one is hard! Any ideas on how we might catch this?

For example, could `cargo-semver-checks` somehow work together with derive macro maintainers — say, have the macro maintainer supply an example use of the macro for the purposes of evaluating semver, and have `cargo-semver-checks` diff the old and new example uses?

**Comment by weiznich** [comment]
For derives with existing documentation examples `cargo-semver-checks` could try building the examples and inspect the generated code there to see if there is an incompatible difference.

**Comment by obi1kenobi** [comment]
Good idea! The following invocation works and is able to generate rustdoc for all examples:
```
RUSTDOCFLAGS="-Z unstable-options --output-format=json" cargo doc --no-deps --examples
```

Using `--example` allows one to specify a specific example to document, instead of going through all.
